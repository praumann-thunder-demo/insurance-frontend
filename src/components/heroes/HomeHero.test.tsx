import { describe, test } from "vitest"
import { render, screen } from "@testing-library/react"
import React from "react"
import HomeHero from "./HomeHero"
import { BrowserRouter } from "react-router-dom"

describe("HomeHero Tests", () => {
  test("Should include header", async () => {
    render(
      <BrowserRouter>
        <HomeHero />
      </BrowserRouter>
    )

    expect(screen.getByText(/Get started/i)).toBeDefined()
    expect(screen.getByText(/Explore plans/i)).toBeDefined()
  })

  test("HomeHero snapshot", async () => {
    const { asFragment } = render(
      <BrowserRouter>
        <HomeHero />
      </BrowserRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
