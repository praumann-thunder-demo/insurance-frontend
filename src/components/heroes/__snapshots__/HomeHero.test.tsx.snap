// Vitest Snapshot v1

exports[`HomeHero Tests > HomeHero snapshot 1`] = `
<DocumentFragment>
  <div
    class="relative"
  >
    <div
      class="absolute inset-x-0 bottom-0 h-1/2 bg-gray-100"
    />
    <div
      class="max-w-7xl mx-auto sm:px-6 lg:px-8"
    >
      <div
        class="relative shadow-xl sm:rounded-2xl sm:overflow-hidden"
      >
        <div
          class="absolute inset-0"
        >
          <img
            alt="Man with grand child"
            class="h-full w-full object-cover"
            src="https://images.unsplash.com/photo-1567067974934-75a3e4534c14?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=2830&q=80&sat=-100"
          />
          <div
            class="absolute inset-0 bg-blue-700 mix-blend-multiply"
          />
        </div>
        <div
          class="relative px-4 py-16 sm:px-6 sm:py-24 lg:py-32 lg:px-8"
        >
          <h1
            class="text-center text-4xl font-extrabold tracking-tight sm:text-5xl lg:text-6xl"
          >
            <span
              class="block text-white"
            >
              Get better care
            </span>
            <span
              class="block text-blue-200 font-semibold"
            >
              It is your life
            </span>
          </h1>
          <p
            class="mt-6 max-w-lg mx-auto text-center text-xl text-blue-200 sm:max-w-3xl"
          >
            When an issue arises, the last thing you want to think about are finances. Our digital tools and large network ensure you get the help you need without worry.
          </p>
          <div
            class="mt-10 max-w-sm mx-auto sm:max-w-none sm:flex sm:justify-center"
          >
            <div
              class="space-y-4 sm:space-y-0 sm:mx-auto sm:inline-grid sm:grid-cols-2 sm:gap-5"
            >
              <a
                class="flex items-center justify-center px-4 py-3 border border-transparent text-base font-medium rounded-md shadow-sm bg-white text-blue-700 hover:bg-blue-50 sm:px-8"
                href="/signup"
              >
                Get started
              </a>
              <a
                class="flex items-center justify-center px-4 py-3 border border-transparent text-base font-medium rounded-md shadow-sm text-white bg-blue-500 bg-opacity-60 hover:bg-opacity-70 sm:px-8"
                href="/plans"
              >
                Explore plans
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
